pkg/util/auth.go

```diff
@@ -0,0 +1,19 @@
+package util
+
+import (
+	"crypto/subtle"
+	macaron "gopkg.in/macaron.v1"
+)
+
+// BasicAuthenticated parses the provided HTTP request for basic authentication credentials
+// and returns true if the provided credentials match the expected username and password.
+// Returns false if the request is unauthenticated.
+// Uses constant-time comparison in order to mitigate timing attacks.
+func BasicAuthenticatedRequest(req macaron.Request, expectedUser, expectedPass string) bool {
```

Comment:
I don't think this function belongs in the util package. Since its only have one caller. Lets put it in the api packages instead. It would also like some tests for this.

Reply:
I moved this to the `api` package and added some tests that verify the basic authentication functionality. Is it worth also adding an end-to-end test that spins up an actual HTTP server?